lib/fs.js

```diff
@@ -1160,7 +1160,7 @@ function FSWatcher() {
   this._handle.onchange = function(status, event, filename) {
     if (status < 0) {
       self._handle.close();
-      self.emit('error', errnoException(status, 'watch'));
+      self.emit('error', errnoException(status, 'watch ' + filename));
```

Comment:
I'd prefer both as I want an uncaught error to still display the filename in the message